### PR TITLE
MongoDB Subscription Consumer feature

### PR DESCRIPTION
**Pull request by rosamarsky**

### CODE REVIEW

File: pkg/mongodb/MongodbMessage.php [hunk 1]
```diff
@@ -65,6 +65,22 @@ public function __construct(string $body = '', array $properties = [], array $he
         $this->redelivered = false;
     }
 
+    public static function fromArrayDbResult(array $arrayResult): self
```
Comment (makasim) [inline pkg/mongodb/MongodbMessage.php]: move the method to the context, make it public with `@internal` docblock, look at AMQP cotext

Comment (makasim) [reply pkg/mongodb/MongodbMessage.php]:
```suggestion
    public static function fromArrayDbResult(array $arrayResult): self
```

Comment (rosamarsky) [reply pkg/mongodb/MongodbMessage.php]: ok, done!

File: pkg/mongodb/MongodbContext.php [hunk 2]
```diff
@@ -107,7 +106,26 @@ public function close(): void
...
-        throw SubscriptionConsumerNotSupportedException::providerDoestNotSupportIt();
+        return new MongodbSubscriptionConsumer($this);
+    }
+
+    /**
+     * @internal It must be used here and in the consumer only
+     */
+    public function convertMessage(array $mongodbMessage): MongodbMessage
+    {
+        $mongodbMessageObj = new MongodbMessage(
```
Comment (makasim) [inline pkg/mongodb/MongodbContext.php]: $this->createMessage()

**Comment by makasim** [comment]
could you please also update the transport doc like this one https://github.com/php-enqueue/enqueue-dev/blob/master/docs/transport/amqp.md#subscription-consumer

**Pull request merged**